jataware/dojo-deprecated
# Add UAZ ontologies when creating an indicator in Dojo.

What's New:
Call UAZ API to fetch and then write into the indicator json prior to creating the indicator

Rationale:
Required to incorporate UAZ work on grounded ontologies and pass along to Causemos

Note: 
- This PR has schema aligned with latest released from Uncharted
- UAZ Ontologies are not currently returned for `qualifier_outputs`; request is in with UAZ to add this functionality

Testing:
If you want a sample json to create an indicator go[ HERE](https://github.com/uncharted-causemos/docs/blob/master/datacubes/indicator.example.json) and copy the example json: 

Then, DELETE (first line):
"$schema": "./indicator.schema.json",

AND...REPLACE:
```
"ontologies": {
    "concepts": [
        {
            "name": "wm/concept/humanitarian_assistance/food_aid",
            "score": 0.785829484462738
        }
    ],
    "processes": [
        {
            "name": "wm/process/train/agriculture_training",
            "score": 0.5917248129844666
        }
    ],
    "properties": [
        {
            "name": "wm/property/condition",
            "score": 0.6441149115562439
        }
    ]
}
```

With: 
`"ontologies": {}`
          
Then:
- Go to local Dojo create indicator and paste in amended json and create the indicator.
- Verify in ES that ontologies are populated
- Go to UAZ API: http://linking.cs.arizona.edu/api#/default/put_v1_groundIndicator
    - Verify ontologies mapped and written to ES match UAZ outputs
    - Note the ontologies in Uncharted's schema example don't match the output, but do match UAZ's output.

Extra Credit: Add another `output` to `outputs` and verify that each output is populated with the correct UAZ ontology information
